src/main/java/appeng/core/sync/packets/PacketNEIRecipe.java

```diff
@@ -62,117 +61,297 @@
+	// }
+	// }
+	// con.onCraftMatrixChanged( craftMatrix );
+	// }
+	// }
+	// }
+	// }
+	// }
+	// }
+
+	/**
+	 * Tries to extract an item from the player inventory. Does account for fuzzy items.
+	 *
+	 * @param player the {@link EntityPlayer} to extract from
+	 * @param mode the {@link Actionable} to simulate or modulate the operation
+	 * @param patternItem which {@link ItemStack} to extract
+	 * @return null or a found {@link ItemStack}
+	 */
+	private ItemStack extractItemFromPlayerInventory( EntityPlayer player, Actionable mode, ItemStack patternItem )
+	{
+		final InventoryAdaptor ia = InventoryAdaptor.getAdaptor( player, ForgeDirection.UNKNOWN );
+		final AEItemStack request = AEItemStack.create( patternItem );
+		final boolean isSimulated = mode == Actionable.SIMULATE;
+		final boolean checkFuzzy = request.isOre() || patternItem.getItemDamage() == OreDictionary.WILDCARD_VALUE || patternItem.hasTagCompound() || patternItem.isItemStackDamageable();
+
+		if( !checkFuzzy )
 		{
-			this.recipe = new ItemStack[9][];
-			for( int x = 0; x < this.recipe.length; x++ )
+			if( isSimulated )
+			{
+				return ia.simulateRemove( 1, patternItem, null );
+			}
+			else
+			{
+				return ia.removeItems( 1, patternItem, null );
+			}
+		}
+		else
+		{
+			if( isSimulated )
 			{
-				NBTTagList list = comp.getTagList( "#" + x, 10 );
+				return ia.simulateSimilarRemove( 1, patternItem, FuzzyMode.IGNORE_ALL, null );
+			}
+			else
+			{
+				return ia.removeSimilarItems( 1, patternItem, FuzzyMode.IGNORE_ALL, null );
+			}
+		}
+	}
+
+	@Override
+	public void fromBytes( ByteBuf buf )
+	{
+		final NBTTagCompound recipe = ByteBufUtils.readTag( buf );
+
+		if( recipe != null )
+		{
+			this.recipeStack = new ItemStack[9][];
+			for( int x = 0; x < this.recipeStack.length; x++ )
+			{
+				final NBTTagList list = recipe.getTagList( "#" + x, 10 );
 				if( list.tagCount() > 0 )
 				{
-					this.recipe[x] = new ItemStack[list.tagCount()];
+					this.recipeStack[x] = new ItemStack[list.tagCount()];
 					for( int y = 0; y < list.tagCount(); y++ )
 					{
-						this.recipe[x][y] = ItemStack.loadItemStackFromNBT( list.getCompoundTagAt( y ) );
+						this.recipeStack[x][y] = ItemStack.loadItemStackFromNBT( list.getCompoundTagAt( y ) );
 					}
 				}
 			}
 		}
 	}
 
-	// api
-	public PacketNEIRecipe( NBTTagCompound recipe ) throws IOException
+	@Override
+	public void toBytes( ByteBuf buf )
 	{
-		ByteBuf data = Unpooled.buffer();
-
-		ByteArrayOutputStream bytes = new ByteArrayOutputStream();
-		DataOutputStream outputStream = new DataOutputStream( bytes );
-
-		data.writeInt( this.getPacketID() );
-
-		CompressedStreamTools.writeCompressed( recipe, outputStream );
-		data.writeBytes( bytes.toByteArray() );
-
-		this.configureWrite( data );
+		ByteBufUtils.writeTag( buf, this.recipe );
+		// ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+		// DataOutputStream outputStream = new DataOutputStream( bytes );
+		//
+		// try
+		// {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)